packages/eslint-plugin/lib/rules/no-nested-await.js

```diff
@@ -62,10 +62,10 @@ module.exports = {
         ) {
           // It's an assignment, so look up to the assigment's parent.
           parent = parent.parent;
-          if (parent.type === 'ExpressionStatement') {
-            // Try to find the parent block.
-            parent = parent.parent;
-          }
+        }
+        if (parent.type === 'ExpressionStatement') {
+          // Try to find the parent block.
+          parent = parent.parent;
```

Comment:
What does this modification tolerate that was not previously tolerated? What’s not tolerated now? What guarantees in the shape of a JS AST does this logic depend upon?

I’m curious how this change composes with the subsequent parent-walk for block statements.

Reply:
Basically, any expression statement should not be considered a "parent".  This accommodates any stray expression statements, like `(await 123);`